### PR TITLE
Bump to kubernetes-1.17.0-rc.2

### PR DESCRIPTION
**Pull request by tkashem**
This moves api to `kubernetes-1.17.0-rc.2`.

Executed the following steps:
- Update `glide.yaml`, bump to `kubernetes-1.17.0-rc.2`
- Run `make build` ( no error )
- Run `make  test` ( no failed tests )
- Run `make verify-scripts` (no changes)

### CODE REVIEW

File: vendor/github.com/golang/protobuf/jsonpb/jsonpb.go [hunk 1]
```diff
@@ -182,7 +183,12 @@ func (m *Marshaler) marshalObject(out *errWriter, v proto.Message, indent, typeU
 				return fmt.Errorf("failed to marshal type URL %q to JSON: %v", typeURL, err)
 			}
 			js["@type"] = (*json.RawMessage)(&turl)
-			if b, err = json.Marshal(js); err != nil {
```
Comment (sttts) [inline vendor/github.com/golang/protobuf/jsonpb/jsonpb.go]: we usually call they commits `bump(*): kubernetes 1.17-rc.2`

**Comment by sttts** [comment]
Only the commit message nit, otherwise lgtm.

File: vendor/github.com/golang/protobuf/jsonpb/jsonpb.go (hunk 1 above)
Comment (tkashem) [reply vendor/github.com/golang/protobuf/jsonpb/jsonpb.go]: @sttts I updated the commit messages, please let me know if you have any further suggestions.

File: glide.lock [hunk 2]
```diff
@@ -1,5 +1,5 @@
-hash: 0cc412c8ccbe6bfefac4bc281cef4d2836b1845b7e1fd9b17497229851380eaa
```
Comment (sttts) [inline glide.lock]: the lock file change should be in the bump commit

Comment (tkashem) [reply glide.lock]: commit: `bump(*): kubernetes 1.17-rc.2`
files: `glide.yaml` and `glide.lock`

commit: `update vendor folder`

commit: fix `make verify`

Is this what you are referring to @sttts ?

Comment (sttts) [reply glide.lock]: Would just do two commits:

- bump(*): kubernetes 1.17-rc.2
- update generated files

Alternatively, sometimes easier to review:

- Pin kube dependencies to 1.17-rc.2
- bump(*)
- update generated files

In both cases the glide.lock should be in the same commit vendor is bumped.

Comment (tkashem) [reply glide.lock]: done, please take a look when you have time @sttts

**Pull request merged**